src/Data/Common/PrimaryFeatureManager.ts

```diff
@@ -18,9 +18,12 @@ import { words } from "@/UI";
 export const PrimaryFeatureManager = (
   jsonParserId: JsonParserId = "Native",
   commitHash: string = "",
-  appVersion: string = "",
-  features?: Pick<ServerStatus, "features" | "extensions" | "version" | "edition" | "slices"> | null
+  appVersion: string = ""
 ): FeatureManager => {
+  const [features, setFeatures] = useState<
+    Pick<ServerStatus, "features" | "extensions" | "version" | "edition" | "slices"> | undefined
```

Comment:
we need the undefined in the typing because we are setting it to undefined ourselves? If that's the case, I'd opt for null, as null shows it's intentionally set to null.